src/module.c

```diff
@@ -532,19 +531,17 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
         while (SeriesIteratorGetNext(&iterator, &sample) != 0 &&
                     (maxResults == -1 || arraylen < maxResults)) {
             RedisModule_ReplyWithArray(ctx, 2);
-
             RedisModule_ReplyWithLongLong(ctx, sample.timestamp);
             RedisModule_ReplyWithDouble(ctx, sample.value);
             arraylen++;
         }
     } else {
         while (SeriesIteratorGetNext(&iterator, &sample) != 0 &&
                     (maxResults == -1 || arraylen < maxResults)) {
-            timestamp_t current_timestamp = sample.timestamp - (sample.timestamp % time_delta);
-            if (current_timestamp > last_agg_timestamp) {
+            if (sample.timestamp >= last_agg_timestamp + time_delta) {
                 ReplyWithAggValue(ctx, last_agg_timestamp, aggObject, context);
                 arraylen++;
-                last_agg_timestamp = current_timestamp;
+                last_agg_timestamp = sample.timestamp - ((sample.timestamp - last_agg_timestamp) % time_delta);
```

Comment:
```
start_ts = 1 
time_delta = 4
last_agg_timestamp = 1
==== 
sample.timestamp = 5
last_agg_timestamp = 1
5 >= 1 + 4 // true
ReplyWithAggValue(ctx, 1, aggObject, context);
// ((sample.timestamp - last_agg_timestamp) % time_delta) is always 0, since sample.timestamp - last_agg_timestamp is always equal to time_delta
last_agg_timestamp = 5 - (( 5 -1 ) % 4 ); // 5 - (4 % 4) = 5 - 0 = 5 
```
We can replace:
```
last_agg_timestamp = sample.timestamp - ((sample.timestamp - last_agg_timestamp) % time_delta);
```
with:
```
last_agg_timestamp = sample.timestamp
```
do you agree?

Reply:
@filipecosta90 
This will reset the baseline and will inconsistent replies.
 
`[[2L, '3'], [12L, '3'], [20L, '3']]` vs. `[[2L, '3'], [10L, '3'], [18L, '3']]`